packages/core/src/types/IViewport.ts

```diff
@@ -6,19 +6,99 @@ import { ActorEntry } from './IActor';
  */
 export type TargetSpecifier = {
   /** The slice index within the current viewport camera to get a reference for */
-  sliceIndex?: number;
-  /** True to get a frame of reference UID reference instead of a regular image one */
+  sliceIndex?: number | [number, number];
+  /**
+   * true to get a frame of reference UID reference instead of a regular image one,
+   * or false to not get it when it would otherwise do so.
+   * The default depends on the view type, but in general showing multiple images
+   * combined, the return will be only for frame of reference, while showing a
+   * single one, it will be for frame of reference.
+   */
   forFrameOfReference?: boolean;
   /** Set of points to get a reference for, in world space */
   points?: Point3[];
   /** The volumeId to reference */
   volumeId?: string;
+  /**
+   * true to include the bounds, if it would not otherwise do so, or
+   * false to exclude them if it would normally do so.
+   * Bounds are normally only included if the viewport is very magnified
+   * AND the points are included in the request, or the view is for a range
+   * of slices.
+   */
+  includeBounds?: boolean;
+};
+
+export type ViewCompatibleOptions = {
+  /** Indicate if the test should proceed if the view were navigated */
+  withNavigation?: boolean;
+  /**
+   * Test to see if this view could be shown in the volume equivalent (eg by
+   * changing the normal etc)
+   */
+  asVolume?: boolean;
+  /**
+   * Use this imageURI for testing - may or may not be the current one.
+   * Should be a straight contains URI for the set of imageIds in any of
+   * the volumes or set of image ids.
+   */
+  imageURI?: string;
+};
+
+/**
+ * A view target is used to get a reference to a view for a viewport.  It
+ * basically says would this viewport show this view, or can direct a viewport
+ * to show a specific view.
+ */
+export type ViewTarget = {
+  /**
+   * The FrameOfReferenceUID
+   */
+  FrameOfReferenceUID: string;
+  /**
+   * An optional property used to specify a slice that this view includes.
+   * For volumes, that will specify which volumes the view applies to, without
+   * necessarily specifying the volumeId.
+   */
+  referencedImageId?: string;
+
+  /**
+   * The position of the camera in world space
+   */
+  cameraPosition?: Point3;
+  /**
+   * The focal point of the camera in world space
+   */
+  cameraFocalPoint?: Point3;
+  /**
+   * The normal for the current view
+   */
+  viewPlaneNormal?: Point3;
+  /**
+   * The viewUp for the view position
+   */
+  viewUp?: Point3;
+  /**
+   * The slice index or range for this view
+   */
+  referencedSliceIndex?: number | [number, number];
+
+  /**
+   * VolumeId that the referencedImageId was chosen from
+   */
+  volumeId?: string;
+  /**
+   * The bounds that are shown.  Allows specifying whether a view includes
+   * particular bounds or not.  This will be in world coordinates.
+   */
+  bounds?: BoundsLPS;
 };
```

Comment:
I like the where these changes are going, but coming to a conclusion that `target` might not be the best naming choice and `reference` is better. We already have referencedImageId, referencedImageIds and referencedVolumeId, referencedSliceIndex. 

Also if we use reference then `forXYZ` prefix becomes redundant as it is implied. Also if you notice 
in your documentation of types you use `to reference...`, `to get a reference for...` a lot which means the explanation
is clearer with `reference` than `target`. e.g., you say `A view target is used to get a reference to a view for a viewport` so ...

So I would suggest we switch from 

`TargetSpecifier` -> `ViewReferenceSpecifier` or `viewReference` then you can rename forFrameOfReference to simple FrameOfReference.

`ViewTarget` -> `ViewReference` or `ViewportReference`

---

also `getViewTarget` to `getViewReference`, since we introduced this recently i think it is fine to change that